runtime/src/main/java/org/corfudb/runtime/clients/LogUnitClient.java

```diff
@@ -127,31 +127,56 @@ public Integer getPort() {
 
     /**
      * Asynchronously read from the logging unit.
+     * Read result is cached at log unit server.
      *
-     * @param address The address to read from.
+     * @param address the address to read from.
      * @return a completableFuture which returns a ReadResponse on completion.
      */
     public CompletableFuture<ReadResponse> read(long address) {
+        return read(address, true);
+    }
+
+    /**
+     * Asynchronously read from the logging unit.
+     *
+     * @param address the address to read from.
+     * @param cacheable whether the read result should be cached on log unit server.
+     * @return a completableFuture which returns a ReadResponse on completion.
+     */
+    public CompletableFuture<ReadResponse> read(long address, boolean cacheable) {
         Timer.Context context = getTimerContext("read");
         CompletableFuture<ReadResponse> cf = sendMessageWithFuture(
-                CorfuMsgType.READ_REQUEST.payloadMsg(new ReadRequest(address)));
+                CorfuMsgType.READ_REQUEST.payloadMsg(new ReadRequest(address, cacheable)));
 
         return cf.thenApply(x -> {
             context.stop();
             return x;
         });
     }
 
+
     /**
      * Read data from the log unit server for a list of addresses.
+     * Read results are <b>NOT</b> cached at log unit server.
      *
      * @param list list of global addresses.
      * @return a completableFuture which returns a ReadResponse on completion.
      */
     public CompletableFuture<ReadResponse> readAll(List<Long> list) {
+       return readAll(list, false);
```

Comment:
can we rename list to addresses or something more verbose.

Reply:
Done.